File: client/src/models/Restaurant.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";

export class Restaurant extends DatabaseItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.visits = data.visits
    this.isShutdown = data.isShutdown
    this.creatorId = data.creatorId
    this.owner = new Profile(data.owner)
    this.reportCount = data.reportCount
  }
}

File: server/Models/Restaurant.cs
using System.ComponentModel.DataAnnotations;

namespace help.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  [Url, MaxLength(1000)] public string ImgUrl { get; set; }
  [MaxLength(3000)] public string Description { get; set; }
  public int Visits { get; set; }
  public bool? IsShutdown { get; set; }
  public string CreatorId { get; set; }
  public Profile Owner { get; set; }
}
